pkg/pipelines/environment.go

```diff
@@ -0,0 +1,84 @@
+package pipelines
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/openshift/odo/pkg/pipelines/meta"
+	"github.com/openshift/odo/pkg/pipelines/roles"
+)
+
+const (
+	envNamespace   = "namespace.yaml"
+	envRoleBinding = "rolebinding.yaml"
+)
+
+// EnvParameters encapsulates parameters for add env command
+type EnvParameters struct {
+	EnvName    string
+	Output     string
+	Prefix     string
+	GitOpsRepo string
+}
+
+// Env will bootstrap a new environment directory
+func Env(o *EnvParameters) error {
+
+	gitopsName := getGitopsRepoName(o.GitOpsRepo)
+	gitopsPath := filepath.Join(o.Output, gitopsName)
+
+	envName := addPrefix(o.Prefix, o.EnvName)
+	envPath := getEnvPath(gitopsPath, o.EnvName, o.Prefix)
+
+	// check if the gitops dir exists
+	exists, _ := isExisting(gitopsPath)
```

Comment:
This doesn't quite do what the comment says, it only checks to see if `gitopsPath` is a valid directory entry, it doesn't check to see whether or not it's a directory or file.